src/js/containers/covid19/homepage/CovidHighlights.jsx

```diff
@@ -0,0 +1,365 @@
+import React, { useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { TooltipWrapper } from 'data-transparency-ui';
+import { get, uniqueId, delay, throttle } from 'lodash';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import moment from 'moment';
+
+import { setOverview } from 'redux/actions/covid19/covid19Actions';
+import CovidOverviewModel from 'models/v2/covid19/BaseOverview';
+import { fetchCovidTotals, fetchDisasterSpending } from 'helpers/disasterHelper';
+import { formatMoneyWithPrecision, calculateUnitForSingleValue } from 'helpers/moneyFormatter';
+import { scrollToY } from 'helpers/scrollToHelper';
+import HeroButton from 'components/homepage/hero/HeroButton';
+
+const TooltipContent = () => (
+    <div className="homepage__covid-19-tt">
+        <h2 className="tooltip__title">COVID 19 Spending Total</h2>
+        <div className="tooltip__text">
+            <p>Tooltip content TBD.</p>
+        </div>
+    </div>
+);
+
+const getTotalSpendingAbbreviated = (totalSpending) => {
+    const unit = calculateUnitForSingleValue(totalSpending);
+    const abbreviatedValue = formatMoneyWithPrecision(totalSpending / unit.unit, 2);
+    return `${abbreviatedValue} ${unit.longLabel}`;
+};
+
+const allDefCAwardTypeCodes = ['02', '03', '04', '05', '06', '07', '08', '09', '10', '11'];
+
+const defaultParams = {
+    filter: {
+        def_codes: ["L", "M", "N", "O", "P"],
+        award_type_codes: allDefCAwardTypeCodes
+    },
+    pagination: {
+        page: 1,
+        limit: 100,
+        order: "desc",
+        sort: "outlay"
+    },
+    spending_type: "total"
+};
+
+let scrollInterval = null;
+let amountUpdate = null;
+
+const updateSpeedByIndexRange = (index) => {
+    const indexAsInt = index + 1;
+    if (indexAsInt <= 100) return 1;
+    if (indexAsInt <= 200) return 2;
+    if (indexAsInt <= 300) return 4;
+    if (indexAsInt <= 400) return 8;
+    if (indexAsInt <= 425) return 16;
+    if (indexAsInt <= 490) return 32;
+    return 64;
+};
+
+const TotalAmount = ({
+    total,
+    isLoading,
+    className,
+    completeIncrement
+}) => {
+    const ref = useRef(null);
+
+    useEffect(() => {
+        const now = moment();
+        const updateAmount = (amount, speedOfUpdate) => new Promise((resolve) => {
+            amountUpdate = delay(() => {
+                ref.current.innerHTML = getTotalSpendingAbbreviated(amount);
```

Comment:
any reason we are specifically using the DOM to get this data instead of keeping it in state?

Reply:
Good question. In this case we're updating the DOM 500 times within six seconds. I think it's more performant to update the DOM directly. I came up with this after looking at some various packages like this one. https://inorganik.github.io/countUp.js/